org.eclipse.scout.sdk.core.typescript/src/main/java/org/eclipse/scout/sdk/core/typescript/builder/imports/ES6ImportValidator.java

```diff
@@ -54,6 +55,9 @@ protected String computeUniqueNameAndRegisterUsage(IDataType type) {
     if (isBuiltInType(name)) {
       return name; // no import required
     }
+    if (type instanceof IES6Class es6Class) {
+      type = es6Class.withoutTypeArguments();
+    }
```

Comment:
Can the typeArguments really be ignored here? Or should they be processed (create imports if required) as well?

Reply:
This is just the computation of a unique name for the type. The type arguments are ignored for this computation. So the unique name of a class with type arguments or the same class without type arguments are  the same.
But the problem here is that the type is used as a key in a map of all unique names. Therefore, the types `SmartField` and `SmartField<number>` will result in the names `SmartField` and `SmartField0`. Omitting the type arguments ensures that only the main type is processed, which is the same in this example and it results only in `SmartField`.

But you are right, the type arguments need to be processed as well. This happens in `DataTypeNameEvaluator#nameForSimpleDataType`, which results in unique names and imports for every type argument.